src/integration/java/io/pinecone/integration/controlPlane/pod/CollectionTest.java

```diff
@@ -20,65 +20,81 @@
 import static io.pinecone.helpers.IndexManager.waitUntilIndexIsReady;
 import static io.pinecone.helpers.IndexManager.createCollection;
 import static io.pinecone.helpers.BuildUpsertRequest.*;
+import static io.pinecone.helpers.AssertRetry.assertWithRetry;
 import static org.junit.jupiter.api.Assertions.*;
 
 @Disabled("Disable the entire class")
 public class CollectionTest {
-
-    private static PineconeControlPlaneClient controlPlaneClient;
     private static final String indexName = RandomStringBuilder.build("collection-test", 8);
+    private static final String collectionName = RandomStringBuilder.build("collection-test", 8);
     private static final ArrayList<String> indexes = new ArrayList<>();
-    private static final ArrayList<String> collections = new ArrayList<>();
     private static final IndexMetric indexMetric = IndexMetric.COSINE;
     private static final List<String> upsertIds = Arrays.asList("v1", "v2", "v3");
     private static final String namespace = RandomStringBuilder.build("ns", 8);
     private static final String apiKey = System.getenv("PINECONE_API_KEY");
     private static final String environment = System.getenv("PINECONE_ENVIRONMENT");
     private static final int dimension = 4;
-
     private static final Logger logger = LoggerFactory.getLogger(CollectionTest.class);
+    private static PineconeControlPlaneClient controlPlaneClient;
+    private static CollectionModel collection;
 
     @BeforeAll
     public static void setUp() throws InterruptedException {
         controlPlaneClient = new PineconeControlPlaneClient(apiKey);
-        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().pods(1).podType("p1.x1").replicas(1).environment(environment);
+
+        // Create and upsert to index
+        CreateIndexRequestSpecPod podSpec =
+                new CreateIndexRequestSpecPod().pods(1).podType("p1.x1").replicas(1).environment(environment);
         CreateIndexRequestSpec spec = new CreateIndexRequestSpec().pod(podSpec);
-        PineconeConnection dataPlaneConnection = createNewIndexAndConnect(controlPlaneClient, indexName, dimension, indexMetric, spec);
+        PineconeConnection dataPlaneConnection = createNewIndexAndConnect(controlPlaneClient, indexName, dimension,
+                indexMetric, spec);
         VectorServiceGrpc.VectorServiceBlockingStub blockingStub = dataPlaneConnection.getBlockingStub();
         indexes.add(indexName);
 
-        blockingStub.upsert(buildRequiredUpsertRequestByDimension(upsertIds, dimension, namespace));
+        // Sometimes we see grpc failures when upserting so quickly after creating, so retry
```

Comment:
This seems to help some of the flake we see when trying to upsert to a fresh index.

Reply:
Is it necessary to upsert data here?

Reply:
I'm creating a new pod index for these tests rather than using `createNewIndexAndConnect()` because I need to use the index name and metric in my tests, which was not easy to do with `createIndexIfNotExistsDataPlane()`.